src/libs/ua.ts

```diff
@@ -1,4 +1,4 @@
-const ua = window.navigator.userAgent.toLowerCase()
+const ua = typeof window !== `undefined` ? window.navigator.userAgent.toLowerCase() : ''
```

Comment:
```suggestion
const ua = typeof window !== 'undefined' ? window.navigator.userAgent.toLowerCase() : ''
```

超絶細かいですが、
変数展開がない純粋な文字列はシングルクオートで統一されてそうです！

Reply:
それとこれは IMHO です！

`const ua = ''` のときでも `ua` を参照する下位のユーティリティ関数が、何かしらの値を返すと思うんですが、
それによって予期せぬ動きとならないように、

```ts
const ua = typeof window !== 'undefined'`
```

 のときは、

```ts
const isId = false
const isSp = false
const isTablet = false
const isPc = true
const isTouchDevice = false
```

みたいな、無難な値を固定値で返すようにしておくとハマりにくいかもしれません！

Reply:
@diescake レビューありがとうございます！！記号の件は直しておきました！
IMHOの件ですが確かに..!
ちょっと考えましたが、そもそも空文字を返してるのがわかりにくいので、'SSR' を返すようにしてみました（もしくは'undefined'とかでも）その値が返ってくるのであれば、isId、isSpも返ってくる値も`false`であると予測がつくかなと。

```
window.navigator !== 'undefined' ? window.navigator.userAgent.toLowerCase() : 'SSR'
```

あとは今回のスコープに含めないですが、useEffect内であればSSR/SSG時でもwindowオブジェクトのエラーが出ないので、最終的には独自フック化してもよいのかなと（そこまでやるかでいえば、まだその時期ではないと思ってますが）
https://stackoverflow.com/questions/59494037/how-to-detect-the-device-on-react-ssr-app-with-next-js

Reply:
横から失礼します！

> useEffect内であればSSR/SSG時でもwindowオブジェクトのエラーが出ないので、最終的には独自フック化してもよいのかな

現象としてはその通りなのですが、 useEffect はアプリケーションのライフサイクルの中で変化する値に対して使うというものという点と、 useEffect はレンダリング後に呼ばれるので最初に ua がない状態のものがレンダリングされてその後に正しい ua を元にしたものが描画されるという現象が起きそうというのがあって、そこから考えると ua を Hooks で取得するのはちょっと違うかもなと思いました！
ただの感想です！ 🙏

Reply:
@nabeliwo 
aaaa、、そもそもトリッキーな方法だということを念頭に置いてなかったで、ご指摘ごもっともです:bow:
現状の回避策も暫定的ではあるので、折をみて詰めていけたらと思います:pray:

Reply:
@kgsi 

あざます！
useEffect については @nabeliwo san と同意見でございますー🙏